src/components/VerticalNavigation/VerticalNavigation.js

```diff
@@ -0,0 +1,252 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import ListGroup from '../ListGroup/ListGroup';
+import VerticalNavigationItem from './VerticalNavigationItem';
+
+// TODO compare with
+// http://www.patternfly.org/pattern-library/navigation/vertical-navigation/
+// TODO fix errors and get it working on storybook
+
+/**
+ * VerticalNavigation - The Vertical Navigation pattern
+ */
+export default class VerticalNavigation extends React.Component {
+  // NOTE: For convenience, ALL PROPS of VerticalNavigation will be passed to all VerticalNavigationItem children.
+  // EXCEPTIONS:
+  // - The children prop, for obvious reasons.
+  // - The depth prop will automatically be set to the currently nested depth (primary, secondary or tertiary).
+  // - Any props passed explicitly to VerticalNavigationItem children will take precendence over inherited props.
+  static renderChildren(props) {
+    const { children, ...parentProps } = props; // TODO maybe take specific props instead of passing all of them?
+    const depth = VerticalNavigation.getNextDepth(props.depth || 'primary');
+    return React.Children.map(children, child => {
+      if (child.type === VerticalNavigationItem) {
+        return React.cloneElement(child, {
+          ...parentProps,
+          depth,
+          ...child.props, // We include this to allow overriding these props inline on a VerticalNavigationItem.
+        });
+      } else {
+        return child;
+      }
+    });
+  }
+
+  static getNextDepth(depth) {
+    return depth === 'primary' ? 'secondary' : 'tertiary';
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      hoverSecondaryNav: false,
+      hoverTertiaryNav: false,
+      collapsedSecondaryNav: false,
+      collapsedTertiaryNav: false,
+    };
+    this.handleNavBarToggleClick = this.handleNavBarToggleClick.bind(this);
+  }
+
+  handleNavBarToggleClick() {
+    const { hideTopBanner } = this.state;
+    this.setState({
+      hideTopBanner: !hideTopBanner,
+    });
```

Comment:
This will need to notify the application as well. The main body content needs to respond to the collapse as well.

Reply:
Good point. I'll add a handler that can be passed in for that.